### PR TITLE
docs: clarify docs for ignore and only params

### PR DESCRIPTION
**Pull request by maiamcc**
a user in #tilt expressed confusion with the docs on `ignore` and `only`
(e.g. they didn't know what directory they're evaluated relative to).
This PR attempts to clarify the API docs and the File Changes Guide
on this point.

### CODE REVIEW

File: api/api.py [hunk 1]
```diff
@@ -97,8 +107,8 @@ def docker_build(ref: str, context: str, build_args: Dict[str, str] = {}, docker
     dockerfile_contents: raw contents of the Dockerfile to use for this build.
     live_update: set of steps for updating a running container (see `Live Update documentation <live_update_reference.html>`_).
     match_in_env_vars: specifies that k8s objects can reference this image in their environment variables, and Tilt will handle those variables the same as it usually handles a k8s container spec's ``image`` s.
-    ignore: set of file patterns that will be ignored. Ignored files will not trigger builds and will not be included in images. Follows the `dockerignore syntax <https://docs.docker.com/engine/reference/builder/#dockerignore-file>`_.
-    only: set of file paths that should be considered for the build. All other changes will not trigger a build and will not be included in images. Inverse of ignore parameter. Only accepts real paths, not file globs.
+    ignore: set of file patterns that will be ignored. Ignored files will not trigger builds and will not be included in images. Follows the `dockerignore syntax <https://docs.docker.com/engine/reference/builder/#dockerignore-file>`_. Patterns will be evaluated relative to ``context``.
```
Comment (landism) [inline api/api.py:110]: maybe s/``context``/the build's ``context``/ here and below? I feel like it'd be easy to wonder if this were referring to some kind of Tiltfile context.

File: docs/custom_build.md [hunk 2]
```diff
@@ -123,6 +123,22 @@ Tilt's `docker_build` supports other options. The most impactful is [Live Update
 
 `custom_build` supports most other options of `docker_build`, and a few specific to non-Docker container builders. If you find an option you think should exist but doesn't, let us know in the `#tilt` channel in [Kubernetes Slack](http://slack.k8s.io).
 
+### Adjust Filewatching wtih `ignore`
```
Comment (landism) [inline docs/custom_build.md:126]:
```suggestion
### Adjust Filewatching with `ignore`
```

**Comment by maiamcc** [comment]
> mind doing `local_resource`, too?
> 
> unfortunately it's not consistent with `custom_build`, and ignores are relative to the Tiltfile dir

le sigh. yes, will do!

File: api/api.py (hunk 1 above)
Comment (maiamcc) [reply api/api.py:110]: good call, changed to "relative to the `context` parameter", better?

**Comment by maiamcc** [comment]
> mind doing `local_resource`, too?
> 
> unfortunately it's not consistent with `custom_build`, and ignores are relative to the Tiltfile dir

...wait this is stupid, shouldn't we just make them the same?

**Comment by landism** [comment]
> ...wait this is stupid, shouldn't we just make them the same?

https://app.clubhouse.io/windmill/story/6088/reconcile-custom-build-and-local-resource-ignore-wd

**Comment by maiamcc** [comment]
Cool, specifying the current local_resource behavior with a mighty hope that we fix it soon 😅

**Pull request merged**